android/authentication.md

```diff
@@ -1512,6 +1512,167 @@ If you want to sign out locally by just deleting tokens, you can call `signOut`
+                        case DONE:
+                            Log.d(TAG, "Sign-in done.");
+                            break;
+                        case SMS_MFA:
+                            Log.d(TAG, "Please confirm sign-in with SMS.");
+                            break;
+                        case NEW_PASSWORD_REQUIRED:
+                            Log.d(TAG, "Please confirm sign-in with new password.");
+                            break;
+                        case CUSTOM_CHALLENGE:
+                            confirmSignIn();
+                            break;
+                        default:
+                            Log.d(TAG, "Unsupported sign-in confirmation: " + signInResult.getSignInState());
+                            break;
+                    }
+                }
+            });
+        }
+
+        @Override
+        public void onError(Exception e) {
+            Log.e(TAG, "Sign-in error", e);
+        }
+    });
+}
+
+public void confirmSignIn() {
+    Map<String, String> res = new HashMap<String, String>();
+    res.put(CognitoServiceConstants.CHLG_RESP_ANSWER, "1133");
+    AWSMobileClient.getInstance().confirmSignIn(res, new Callback<SignInResult>() {
+        @Override
+        public void onResult(final SignInResult signInResult) {
+            runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    Log.d(TAG, "Sign-in callback state: " + signInResult.getSignInState());
+                    switch (signInResult.getSignInState()) {
+                        case DONE:
+                            Log.d(TAG, "Sign-in done.");
+                            break;
+                        case SMS_MFA:
+                            Log.d(TAG, "Please confirm sign-in with SMS.");
+                            break;
+                        case NEW_PASSWORD_REQUIRED:
+                            Log.d(TAG, "Please confirm sign-in with new password.");
+                            break;
+                        default:
+                            Log.d(TAG, "Unsupported sign-in confirmation: " + signInResult.getSignInState());
+                            break;
+                    }
+                }
+            });
+        }
+
+        @Override
+        public void onError(Exception e) {
+            Log.e(TAG, "Confirm Custom auth Sign-in error", e);
+        }
+    });
+}
+```
+
+### Creating a CAPTCHA
+
+Here is the sample for creating a CAPTCHA challenge with a Lambda Trigger.
+
+The `Create Auth Challenge Lambda Trigger` creates a CAPTCHA as a challenge to the user. The URL for the CAPTCHA image and  the expected answer is added to the private challenge parameters:
+
+```javascript
+export const handler = async (event) => {
+    if (!event.request.session || event.request.session.length === 0) {
+        event.response.publicChallengeParameters = {
+            captchaUrl: "url/123.jpg",
```

Comment:
nit: maybe something a little more evocative of a captcha image rather than a random URL, like `/path/to/captchaService/image.jpeg`--the point is that it should be clear the customer needs to replace